lexers/lexer.lua

```diff
@@ -1676,20 +1676,106 @@ M.ignored_extensions = {
 	orig = true, bak = true, old = true, new = true
 }
 
+--- Map of hashbang utilities to associated lexer names.
+-- @usage lexer.detect_utilities.luadoc = 'lua'
+M.detect_utilities = {}
+
 --- Map of first-line patterns to their associated lexer names.
 -- These are Lua string patterns, not LPeg patterns.
 -- @usage lexer.detect_patterns['^#!.+/zsh'] = 'bash'
 M.detect_patterns = {}
 
---- Returns the name of the lexer often associated a particular filename and/or file content.
--- @param[opt] filename String filename to inspect. The default value is read from the
---   "lexer.scintillua.filename" property.
--- @param[optchain] line String first content line, such as a shebang line. The default value
---   is read from the "lexer.scintillua.line" property.
--- @return string lexer name to pass to `lexer.load()`, or `nil` if none was detected
-function M.detect(filename, line)
-	if not filename then filename = M.property and M.property['lexer.scintillua.filename'] or '' end
-	if not line then line = M.property and M.property['lexer.scintillua.line'] or '' end
+--[[ hashbang check
+	hashbangs only have PATH <SPACE> ARG
+	NOTE: #! have a line limit of 256 characters
+		if PATH matches /env, utility should be in args
+			POSIX does not have -S, but BSDs/Linux seem to
+			All do seem to have name=value
+			discard first arg if /-[^S]*S/; and all subsequent /=/
+			NOTE: this means you can't have a command with /^-|=/
+	NOTES: long-options (GNUisms) unsupported
+--]]
+local function GetHashBang(line)
```

Comment:
This seems like an awful lot of work to avoid populating `M.detect_patterns` with more hashbang patterns.

Reply:
Given that env -S allows for variables, you could have a variable called { env -S python_ignore=true awk -f } would lead to python being set as the syntax, a false positive.